jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java

```diff
@@ -58,8 +58,12 @@
       for (LogData logData : logs) {
         Log jLog = new Log();
         jLog.setTimestamp(logData.getTime());
-        final Tag tag = buildTag(logData.getMessage(), logData.getPayload());
-        jLog.setFields(new ArrayList<Tag>() {{add(tag);}});
+        if (logData.getFields() != null) {
+          jLog.setFields(buildTags(logData.getFields()));
+        } else {
+          final Tag tag = buildTag(logData.getMessage(), logData.getPayload());
```

Comment:
I think this is incorrect. Should be `"event": message, "payload": payload` (and drop either if null)